### PR TITLE
upgrade to Ginkgo v2

### PR DESCRIPTION
**Pull request by SamYuan1990**
#### Type of change

- Improvement (improvement to code, performance, etc)
- Test update

#### Description

upgrade ginkgo to v2

### CODE REVIEW

**Comment by SamYuan1990** [comment]
as reopen for https://github.com/hyperledger/fabric/pull/3202

**Comment by SamYuan1990** [comment]
```
2022-06-21T14:14:05.4200851Z coverage: 95.3% of statements
2022-06-21T14:14:05.4201233Z FAIL	github.com/hyperledger/fabric/orderer/common/cluster	67.052s
```
fails from CI but passed locally, can any one help?
image

**Comment by SamYuan1990** [comment]
```
--- FAIL: TestWithStaticDeliverClientLeader (90.03s)
    gossip_service_test.go:630: Failed to establish full channel membership. Only 0 out of 2 peers have full membership
FAIL
coverage: 45.0% of statements
FAIL	github.com/hyperledger/fabric/gossip/service	104.114s
?   	github.com/hyperledger/fabric/gossip/service/mocks	[no test files]
ok  	github.com/hyperledger/fabric/gossip/state	88.116s	coverage: 79.7% of statements
ok  	github.com/hyperledger/fabric/gossip/state/mocks	0.034s	coverage: 55.6% of statements
ok  	github.com/hyperledger/fabric/gossip/util	4.054s	coverage: 73.6% of statements
FAIL
```
from ci https://dev.azure.com/Hyperledger/Fabric/_build/results?buildId=53950&view=logs&j=e306c17a-d139-54bf-a475-f5a11259cee7&t=1e3023a5-584f-52f3-49bc-66bd27d27b6d&l=138
but seems this test case passed at local?
image

**Comment by SamYuan1990** [comment]
hope anyone can help re run the CI to see if it works.
btw I use https://marketplace.visualstudio.com/items?itemName=joselitofilho.ginkgotestexplorer this plugin at local to trigger unit test.

File: go.mod [hunk 1]
```diff
@@ -87,7 +86,7 @@ require (
 	github.com/moby/sys/mountinfo v0.4.0 // indirect
 	github.com/moby/term v0.0.0-20201216013528-df9cb8a40635 // indirect
 	github.com/morikuni/aec v1.0.0 // indirect
-	github.com/nxadm/tail v1.4.8 // indirect
+	github.com/onsi/ginkgo/v2 v2.1.3
```
Comment (denyeart) [inline go.mod]: Since ginkgo is a direct dependency, shouldn't it be in the upper 'require' section?

Comment (SamYuan1990) [reply go.mod]: updated.

**Review by denyeart — APPROVED** [review]
Thanks @SamYuan1990 , this looks good to me. I'm re-running unit test to see if we can get it to pass.

**Pull request merged**